### PR TITLE
feat: exclude bootstrap server for suspended, suspending and resuming…

### PR DESCRIPTION
**Pull request by valeriiashapoval**
## Description
(https://issues.redhat.com/browse/MGDSTRM-9912) 

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
- [X] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [X] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Comment by pawelpaszki** [comment]
changes look good to me. I'll let @JameelB take a look as well

**Comment by valeriiashapoval** [comment]
> changes look good to me. I'll let @JameelB take a look as well

thanks @pawelpaszki !

File: internal/kafka/test/integration/admin_kafka_test.go [hunk 1]
```diff
@@ -1171,6 +1171,7 @@ func TestAdminKafka_Update(t *testing.T) {
 				g.Expect(resp.StatusCode).To(gomega.Equal(http.StatusOK))
 				g.Expect(result.Id).To(gomega.Equal(sampleKafkaID1))
 				g.Expect(result.Status).To(gomega.Equal(constants.KafkaRequestStatusReady.String()))
+				g.Expect(result.BootstrapServerHost).To(gomega.BeEmpty())
```
Comment (machi1990) [inline internal/kafka/test/integration/admin_kafka_test.go:1174]: can we ensure that the bootstrap server host is set in the database for all these kafkas https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f9ab2b6af9c72e956df5e1833a3a2235a1b982aa/internal/kafka/test/integration/admin_kafka_test.go#L1351-L1522

in the meantime it is empty

File: internal/kafka/test/integration/admin_kafka_test.go [hunk 2]
```diff
@@ -1171,6 +1171,7 @@ func TestAdminKafka_Update(t *testing.T) {
 				g.Expect(resp.StatusCode).To(gomega.Equal(http.StatusOK))
```
Comment (machi1990) [inline internal/kafka/test/integration/admin_kafka_test.go:1171]: can also add an integration test from the user facing perspective: there are some tests already in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f9ab2b6af9c72e956df5e1833a3a2235a1b982aa/internal/kafka/test/integration/kafkas_test.go

File: internal/kafka/internal/presenters/admin_kafka.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ func PresentKafkaRequestAdminEndpoint(kafkaRequest *dbapi.KafkaRequest, accountS
 		Region:                     kafkaRequest.Region,
 		Owner:                      kafkaRequest.Owner,
 		Name:                       kafkaRequest.Name,
-		BootstrapServerHost:        setBootstrapServerHost(kafkaRequest.BootstrapServerHost),
+		BootstrapServerHost:        setBootstrapServerHost(kafkaRequest),
```
Comment (machi1990) [inline internal/kafka/internal/presenters/admin_kafka.go:43]: @JameelB @pawelpaszki @MikeEdgar do we want to exclude the bootstrap server host from the admin endpoints as well, just like it is done in this PR? This seems consistent but I just want to be sure if this is something we want to do

File: internal/kafka/internal/presenters/kafka.go [hunk 4]
```diff
@@ -114,11 +115,16 @@ func PresentKafkaRequest(kafkaRequest *dbapi.KafkaRequest, kafkaConfig *config.K
 	}, nil
 }
 
-func setBootstrapServerHost(bootstrapServerHost string) string {
-	if bootstrapServerHost != "" {
-		return fmt.Sprintf("%s:443", bootstrapServerHost)
+func setBootstrapServerHost(request *dbapi.KafkaRequest) string {
+	if request.BootstrapServerHost != "" {
+		switch request.Status {
+		case constants.KafkaRequestStatusSuspended.String(), constants.KafkaRequestStatusSuspending.String(), constants.KafkaRequestStatusResuming.String():
```
Comment (machi1990) [inline internal/kafka/internal/presenters/kafka.go:121]: line 121-122 are not unit tested. Can we ensure we've a test for this?

File: internal/kafka/internal/presenters/admin_kafka.go (hunk 3 above)
Comment (JameelB) [reply internal/kafka/internal/presenters/admin_kafka.go:43]: If we drop the bootstrap_server_host in the public endpoints, I think it make sense to also drop it in the admin endpoint for consistency. Unless there's strong opinions against doing that. 

Note that I have added a topic in our tech discussion if we really want to exclude both these fields from the public/admin APIs.

File: internal/kafka/internal/presenters/kafka.go [hunk 5]
```diff
@@ -114,11 +115,16 @@ func PresentKafkaRequest(kafkaRequest *dbapi.KafkaRequest, kafkaConfig *config.K
...
+			return ""
```
Comment (machi1990) [inline internal/kafka/internal/presenters/kafka.go:122]: I am wondering if we should also update the description: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f9ab2b6af9c72e956df5e1833a3a2235a1b982aa/openapi/kas-fleet-manager.yaml#L981 ? 
@JameelB

Comment (machi1990) [reply internal/kafka/internal/presenters/kafka.go:122]: same for the admin endpoints api

File: internal/kafka/internal/presenters/kafka.go [hunk 6]
```diff
@@ -85,7 +86,7 @@ func PresentKafkaRequest(kafkaRequest *dbapi.KafkaRequest, kafkaConfig *config.K
 		CloudProvider:              kafkaRequest.CloudProvider,
 		MultiAz:                    kafkaRequest.MultiAZ,
 		Owner:                      kafkaRequest.Owner,
-		BootstrapServerHost:        setBootstrapServerHost(kafkaRequest.BootstrapServerHost),
+		BootstrapServerHost:        setBootstrapServerHost(kafkaRequest),
 		AdminApiServerUrl:          kafkaRequest.AdminApiServerURL,
```
Comment (machi1990) [inline internal/kafka/internal/presenters/kafka.go:90]: The JIRA also mentions stop showing `AdminApiServerUrl` depending on the suspending/suspended/resuming state.

File: internal/kafka/test/integration/admin_kafka_test.go (hunk 1 above)
Comment (valeriiashapoval) [reply internal/kafka/test/integration/admin_kafka_test.go]: Thanks @machi1990, just committed the change!

File: internal/kafka/internal/presenters/kafka.go (hunk 4 above)
Comment (valeriiashapoval) [reply internal/kafka/internal/presenters/kafka.go:121]: thanks @machi1990 ! The lines should be covered now. For both bootstrap and admin server url.

File: internal/kafka/internal/presenters/kafka.go (hunk 6 above)
Comment (valeriiashapoval) [reply internal/kafka/internal/presenters/kafka.go]: Thanks @machi1990 . Just added the function to set the Url.

**Comment by pb82** [comment]
@valeriiashapoval the agreement we reached is to always return those URLs when they are known. Otherwise it would make for an unpredictable API. I think we can close this PR.

**Pull request closed without merging**